assessment/collectors/infoschema_collector.go

```diff
@@ -0,0 +1,195 @@
+/* Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.*/
+
+package assessment
+
+import (
+	"database/sql"
+	"fmt"
+
+	common "github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/sources"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/sources/mysql"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/common/constants"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/conversion"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/internal"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/profiles"
+)
+
+type InfoSchemaCollector struct {
+	tables           []utils.TableAssessment
+	indexes          []utils.IndexAssessment
+	triggers         []utils.TriggerAssessment
+	storedProcedures []utils.StoredProcedureAssessment
+	conv             *internal.Conv
+}
+
+func (c InfoSchemaCollector) IsEmpty() bool {
+	if c.conv == nil && c.indexes == nil && c.tables == nil && c.storedProcedures == nil && c.triggers == nil {
+		return true
+	}
+	return false
+}
+
+func CreateInfoSchemaCollector(conv *internal.Conv, sourceProfile profiles.SourceProfile) (InfoSchemaCollector, error) {
+	logger.Log.Info("initializing infoschema collector")
+	dbIdentifier := utils.DbIdentifier{}
+	if sourceProfile.Conn.Ty == profiles.SourceProfileConnectionTypeMySQL {
+		dbIdentifier.DatabaseName = sourceProfile.Conn.Mysql.Db
+	}
+	tb := []utils.TableAssessment{}
+	var errString string
+	for _, table := range conv.SrcSchema {
+		columnAssessments := []utils.ColumnAssessment[any]{}
+		for _, column := range table.ColDefs {
+			columnAssessments = append(columnAssessments, utils.ColumnAssessment[any]{
+				Db:        dbIdentifier,
+				Name:      column.Name,
+				TableName: table.Name,
+				ColumnDef: column,
+			})
+		}
+		tb = append(tb, utils.TableAssessment{Name: table.Name, TableDef: table, ColumnAssessments: columnAssessments, Db: dbIdentifier})
+	}
+	infoSchema, err := getInfoSchema(sourceProfile)
+	if err != nil {
+		return InfoSchemaCollector{}, err
+	}
+	indCollector := []utils.IndexAssessment{}
```

Comment:
nit: moving these into respective methods will make the code more readable and testable.